sklearn/utils/estimator_checks.py

```diff
@@ -1931,6 +1930,8 @@ def check_class_weight_classifiers(name, classifier_orig):
             classifier.set_params(max_iter=1000)
         if hasattr(classifier, "min_weight_fraction_leaf"):
             classifier.set_params(min_weight_fraction_leaf=0.01)
+        if hasattr(classifier, "tol"):
+            classifier.set_params(tol=None)
```

Comment:
I had to add that because with the default tol=1e-3 it would converge in a few iterations (~20) and the score is not as high as expected in the test.

I'm not sure it's the right way to fix it though...